cedar-policy-validator/src/err.rs

```diff
@@ -26,78 +26,82 @@ use thiserror::Error;
-    #[error("Undeclared common types: {0:?}")]
-    UndeclaredCommonType(HashSet<String>),
+    /// Undeclared common type(s) used in entity or context attributes.
+    #[error("undeclared common type(s): {0:?}")]
+    UndeclaredCommonTypes(HashSet<String>),
     /// Duplicate specifications for an entity type. Argument is the name of
     /// the duplicate entity type.
-    #[error("Duplicate entity type {0}")]
+    #[error("duplicate entity type: {0}")]
     DuplicateEntityType(String),
     /// Duplicate specifications for an action. Argument is the name of the
     /// duplicate action.
-    #[error("Duplicate action {0}")]
+    #[error("duplicate action: {0}")]
     DuplicateAction(String),
     /// Duplicate specification for a reusable type declaration.
-    #[error("Duplicate common type {0}")]
+    #[error("duplicate common type: {0}")]
     DuplicateCommonType(String),
     /// Cycle in the schema's action hierarchy.
-    #[error("Cycle in action hierarchy")]
+    #[error("cycle in action hierarchy")]
     CycleInActionHierarchy,
     /// Parse errors occurring while parsing an entity type.
-    #[error("Parse error in entity type: {}", Self::format_parse_errs(.0))]
+    #[error("parse error in entity type: {}", Self::format_parse_errs(.0))]
     EntityTypeParseError(ParseErrors),
     /// Parse errors occurring while parsing a namespace identifier.
-    #[error("Parse error in namespace identifier: {}", Self::format_parse_errs(.0))]
+    #[error("parse error in namespace identifier: {}", Self::format_parse_errs(.0))]
     NamespaceParseError(ParseErrors),
     /// Parse errors occurring while parsing an extension type.
-    #[error("Parse error in extension type: {}", Self::format_parse_errs(.0))]
+    #[error("parse error in extension type: {}", Self::format_parse_errs(.0))]
     ExtensionTypeParseError(ParseErrors),
     /// Parse errors occurring while parsing the name of one of reusable
     /// declared types.
-    #[error("Parse error in common type identifier: {}", Self::format_parse_errs(.0))]
+    #[error("parse error in common type identifier: {}", Self::format_parse_errs(.0))]
     CommonTypeParseError(ParseErrors),
     /// The schema file included an entity type `Action` in the entity type
     /// list. The `Action` entity type is always implicitly declared, and it
     /// cannot currently have attributes or be in any groups, so there is no
     /// purposes in adding an explicit entry.
-    #[error("Entity type `Action` declared in `entityTypes` list")]
+    #[error("entity type `Action` declared in `entityTypes` list")]
     ActionEntityTypeDeclared,
     /// One or more action entities are declared with `attributes`, but this is
     /// not currently supported.
-    #[error("Actions declared with `attributes`: [{}]", .0.iter().join(", "))]
-    ActionEntityAttributes(Vec<String>),
+    #[error("actions declared with `attribute(s)`: [{}]", .0.iter().join(", "))]
```

Comment:
the parens seem weird here, because isn't the actual JSON key `attributes` without parens?

Reply:
Yes, good catch.